Sources/Workspace/Workspace.swift

```diff
@@ -1799,30 +1799,61 @@ extension Workspace {
 
         private let workspace: Workspace
 
+        private let observabilityScope: ObservabilityScope
+
+        private let _dependencies: LoadableResult<(required: OrderedCollections.OrderedSet<PackageReference>, missing: OrderedCollections.OrderedSet<PackageReference>)>
+
+        private let _constraints: LoadableResult<[PackageContainerConstraint]>
```

Comment:
nit: missing doc comments

Reply:
we usually document public APIs or non-trivial ones

Reply:
IMO for internal APIs it would also be useful to clarify the intent. That would make it easier to review this code as a part of this PR or when revisiting it later. Also would help external contributors if those are interested in contributing to this area of code.